pkg/apps/controller/deployer/factory.go

```
@@ -124,6 +126,15 @@ func (c *DeploymentController) updateReplicationController(old, cur interface{})
 	c.enqueueReplicationController(curRC)
 }
 
+func (c *DeploymentController) addPod(obj interface{}) {
+	pod := obj.(*v1.Pod)
+
+	rc, err := c.rcForDeployerPod(pod)
+	if err == nil && rc != nil {
```

i would return an error in case there is no deployer pod for rc